src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,150 @@
+package core.basesyntax;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import core.basesyntax.service.InvalidUserException;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final String INVALID_DATA = "Invalid user data."
+            + " Please check login, password, and age.";
+    private static final String EXISTS_MSG = "User with this login already exists.";
+    private static RegistrationServiceImpl registrationService;
+
+    @BeforeAll
+    public static void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void register_ValidUserRegistration_NotNullAndEquals() {
+        User user = new User();
+        user.setLogin("testuser1");
+        user.setPassword("password1231");
+        user.setAge(20);
+
+        User registeredUser = registrationService.register(user);
+
+        Assertions.assertNotNull(registeredUser);
+        Assertions.assertNotNull(registeredUser.getId());
+        Assertions.assertEquals(user.getLogin(), registeredUser.getLogin());
+        Assertions.assertEquals(user.getPassword(), registeredUser.getPassword());
+        Assertions.assertEquals(user.getAge(), registeredUser.getAge());
+    }
+
+    @Test
+    public void register_InvalidUserRegistration_ExceptionThrown() {
+        User user = new User();
+        user.setLogin("short");
+        user.setPassword("password123");
+        user.setAge(20);
+
+        InvalidUserException exception = Assertions.assertThrows(InvalidUserException.class, () -> {
+            registrationService.register(user);
+        });
+
+        Assertions.assertEquals(INVALID_DATA, exception.getMessage());
```

Comment:
not critical for now, but it's an redundant check - imagine if we want just to rephrase exception message in throwing place. this test would fail, but we don't want to